src/Domain.php

```diff
@@ -651,15 +691,63 @@ public function withoutLabel(int $key, int ...$keys): self
         }
 
         if ([] === $labels) {
-            return new self();
+            return new self(null, null, ...$this->getIDNAOptions());
         }
 
         $domain = implode('.', array_reverse($labels));
         $psContent = $this->publicSuffix->getContent();
         if (null === $psContent || '.'.$psContent !== substr($domain, - strlen($psContent) - 1)) {
-            return new self($domain);
+            return new self($domain, null, ...$this->getIDNAOptions());
         }
 
-        return new self($domain, $this->publicSuffix);
+        return new self($domain, $this->publicSuffix, ...$this->getIDNAOptions());
+    }
+    
+    /**
+     * @return array
+     */
+    public function getIDNAOptions(): array
+    {
+        return [$this->asciiIDNAOption, $this->unicodeIDNAOption];
+    }
+    
+    public function getAsciiIDNAOption(): int
+    {
+        return $this->asciiIDNAOption;
+    }
+    
+    public function getUnicodeIDNAOption(): int
+    {
+        return $this->unicodeIDNAOption;
+    }
+    /**
+     * Set IDNA_* options for functions idn_to_ascii, idn_to_utf.
+     * @see https://www.php.net/manual/en/intl.constants.php
+     * @param  int   $forAscii
+     * @param  int   $forUnicode
+     * @return $this
+     */
+    public function setIDNAOptions(int $forAscii, int $forUnicode)
```

Comment:
this method should be called `withIDNAOptions` and must return a new instance because the Domain object and the PublicSuffix object are Immutable Value Objects